app/api/saml/acs/route.ts

```diff
@@ -0,0 +1,86 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { saml } from '@/lib/saml-config'
+
+export async function POST(request: NextRequest) {
+  try {
+    const formData = await request.formData()
+    const samlResponse = formData.get('SAMLResponse') as string
+
+    if (!samlResponse) {
+      throw new Error('No SAML response received')
+    }
+
+    console.log('🔍 Processing SAML response with @node-saml/node-saml...')
+
+    const { profile } = await saml.validatePostResponseAsync({ SAMLResponse: samlResponse })
+
+    if (!profile) {
+      throw new Error('No profile returned from SAML response')
+    }
+
+    console.log('✅ SAML validation succeeded!')
+    console.log('📋 Profile:', JSON.stringify(profile, null, 2))
+
+    const attributes = (profile.attributes || {}) as Record<string, unknown>
+
+    // Helper function to get attribute value (handles both single values and arrays)
+    const getAttr = (key: string): string | undefined => {
+      const value = attributes[key]
+      if (Array.isArray(value)) {
+        return value[0] as string
+      }
+      return value as string | undefined
+    }
+
+    const user = {
```

Comment:
You probably want to type this user object so that you can use it again. Assuming you are going to put the user in the JWT.